desktop_version/src/Editor.cpp

```diff
@@ -3192,6 +3320,123 @@ void editorclass::get_input_line(const enum TextMode mode, const std::string& pr
     old_entity_text = key.keybuffer;
 }
 
+void process_editor_buffer(const bool undo)
+{
+    extern editorclass ed;
+
+    std::vector<EditorUndoInfo>* buffer = undo ? &ed.undo_buffer : &ed.redo_buffer;
+
+    if (buffer->size() == 0)
+    {
+        ed.show_note(loc::gettext(undo ? "ERROR: Nothing to undo" : "ERROR: Nothing to redo"));
```

Comment:
In general we try to keep all `loc::gettext` strings easily greppable. Please move the `loc::gettext` calls into the ternary, not surrounding it.

Speaking of which, you need to add these strings to the language files too.